src/LockIn-SignalRecovery_7265DSP/main.py

```diff
@@ -505,16 +505,11 @@ def find_best_time_constant_key(self, time_constant):
 
         time_constant = self.value_to_float(time_constant)
 
-        # sum over boolean entries leads to index of new time_constant
-        # this also works if the order of the time constants list is lost as the number of True and False
-        # remains the same
-        tc_index = sum(np.array(self.timeconstants_numbers) < time_constant)  # sum over boolean entries
+        tc_index = sum(np.array(self.timeconstants_numbers) > time_constant)  # sum over boolean entries
```

Comment:
/implement sum over boolean entries leads to index of new time_constant. this also works if the order of the time constants list is lost as the number of True and False remains the same.